test/test_hessian.jl

```diff
@@ -1,15 +1,17 @@
 using SparseConnectivityTracer
-using SparseConnectivityTracer: Dual, HessianTracer, MissingPrimalError, trace_input, empty
+using SparseConnectivityTracer:
+    Dual, HessianTracer, MissingPrimalError, trace_input, empty, isshared
```

Comment:
Can we add a test where we manually create a shared tracer via `create_tracers`, run it through a function with multiple outputs and check for egality of the outputs with `===`? This could ensure we don't accidentally create new tracers somewhere.

Reply:
I don't see how that would safeguard against new tracer creation. And such a test is likely subsumed by the ones I added in the Hessian pattern section

Reply:
I added tests with the function below, but here are a few caveats.

```julia
function multi_output_for_shared_test(x::AbstractArray)
    z = ones(eltype(x), size(x))
    y1 = x[1]^2 * z[1]
    y2 = z[2] * x[2]^2
    y3 = x[1] * x[2]
    y4 = z[1] * z[2]
    return [y1, y2, y3, y4]
end
```

- When two `HessianTracer`s are merged in "shared" mode, it's arbitrary which one gets picked to be mutated because they're _supposed_ to be the same object. So in theory, `y1` and `y2` could end up being different objects.
  - It doesn't happen here because of the emptiness filter, which is applied before the `_inner` routine: `z[1]` and `z[2]` are empty so they're not even considered.
  - It could happen theoretically if a non-empty tracer were created with no relation whatsoever to the initial array. However I'm not sure that's a state our code could ever be in.
- Since the contents of `z` are new tracer objects, the last coordinate `y4` contains a different Hessian object than the rest. I'm not sure that can be solved, except by disallowing `myempty` in "shared" mode as I suggested. Hence the following tests:

```julia
yt = # output of the function on a shared HessianTracer array
@test yt[1].hessian == yt[2].hessian
@test yt[1].hessian == yt[3].hessian
@test_broken yt[1].hessian == yt[4].hessian
```